drivers/rtc/rtc_rv8803.c

```diff
@@ -0,0 +1,861 @@
+		}
+
+		err = gpio_pin_interrupt_configure_dt(&config->gpio_int, GPIO_INT_EDGE_TO_ACTIVE);
+		if (err) {
+			LOG_ERR("Failed to enable GPIO interrupt, error: %d", err);
+			return err;
+		}
+
+		gpio_init_callback(&data->irq_callback, rv8803_irq_handler,
+				   BIT(config->gpio_int.pin));
+
+		err = gpio_add_callback_dt(&config->gpio_int, &data->irq_callback);
+		if (err) {
+			LOG_ERR("Failed to add GPIO callback, error: %d", err);
+			return err;
+		}
+
+		data->dev = dev;
+		data->work.handler = rv8803_work_callback;
+	}
+#endif
+
+	/* Configure CLKOUT frequency */
+	switch (config->clkout_freq) {
+	case RV8803_PROP_ENUM_1HZ:
+		freq = RV8803_EXTENSION_FD_1Hz;
+		break;
+	case RV8803_PROP_ENUM_1024HZ:
+		freq = RV8803_EXTENSION_FD_1024Hz;
+		break;
+	case RV8803_PROP_ENUM_32768HZ:
+	default:
+		freq = RV8803_EXTENSION_FD_32768Hz;
+		break;
+	}
+	err = rv8803_update_reg8(dev, RV8803_EXTENSION_REG, RV8803_EXTENSION_FD_MASK, freq);
+	if (err) {
+		return -ENODEV;
+	}
+
+	/* Clear alarm and update flag */
+	err = rv8803_update_reg8(dev, RV8803_CONTROL_REG, RV8803_FLAG_AF_BIT | RV8803_FLAG_UF_BIT,
+				 RV8803_FLAG_AF_BIT | RV8803_FLAG_UF_BIT);
+	if (err) {
+		return -ENODEV;
+	}
+
+	/* Disable IRQs */
+	err = rv8803_update_reg8(dev, RV8803_CONTROL_REG,
+				 RV8803_CONTROL_AIE_BIT | RV8803_CONTROL_UIE_BIT, 0);
+	if (err) {
+		return -ENODEV;
+	}
+
+	/* Disable alarms */
+	err = rv8803_read_regs(dev, RV8803_MINUTES_ALARM_REG, regs, sizeof(regs));
+	if (err) {
+		return -ENODEV;
+	}
+
+	regs[0] |= RV8803_MINUTES_ALARM_AE_M_BIT;
+	regs[1] |= RV8803_HOURS_ALARM_AE_H_BIT;
+	regs[2] |= RV8803_WEEKDAY_OR_DATE_ALARM_AE_WD_BIT;
+
+	err = rv8803_write_regs(dev, RV8803_MINUTES_ALARM_REG, regs, sizeof(regs));
+	if (err) {
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static const struct rtc_driver_api rv8803_driver_api = {
+	.set_time = rv8803_set_time,
+	.get_time = rv8803_get_time,
+#ifdef CONFIG_RTC_ALARM
+	.alarm_get_supported_fields = rv8803_alarm_get_supported_fields,
+	.alarm_set_time = rv8803_alarm_set_time,
+	.alarm_get_time = rv8803_alarm_get_time,
+	.alarm_is_pending = rv8803_alarm_is_pending,
+	.alarm_set_callback = rv8803_alarm_set_callback,
+#endif
+#if defined(RV8803_INT_GPIOS_IN_USE) && defined(CONFIG_RTC_UPDATE)
+	.update_set_callback = rv8803_update_set_callback,
+#endif
+#ifdef CONFIG_RTC_CALIBRATION
+	.set_calibration = rv8803_set_calibration,
+	.get_calibration = rv8803_get_calibration
+#endif
+};
+
+#define RV8803_INIT(inst)                                                                          \
+	static const struct rv8803_config rv8803_config_##inst = {                                 \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.clkout_freq = DT_INST_ENUM_IDX_OR(inst, clkout_frequency, 0),                     \
+		IF_ENABLED(RV8803_INT_GPIOS_IN_USE,                                                \
+		(.gpio_int = GPIO_DT_SPEC_INST_GET_OR(inst, int_gpios, {0})))};             \
```

Comment:
Please indent the macro argument and move the final closing bracket down:
```
IF_ENABLED(RV8803_INT_GPIOS_IN_USE,
        (.gpio_int = GPIO_DT_SPEC_INST_GET_OR(inst, int_gpios, {0})))
};
```
and align the `\`

I would even expand it even further to:
```
IF_ENABLED(
        RV8803_INT_GPIOS_IN_USE,
        (.gpio_int = GPIO_DT_SPEC_INST_GET_OR(inst, int_gpios, {0}))
)
};
```
to make the macro as readable as possible :)

Reply:
I've tried to fix that too, but after the change `check_compliance.py` keeps complaining about line 860. I don't really understand why. Running `clang-format` on the file fixes that error, but causes the backslash in that line to be misaligned

Reply:
Likely a trailing space, or space before tab or something, do you have whitespace rendering enabled in your IDE?